File: engine/engine-model/src/main/scala/tools/packet/serverlist/GetServerStatus.scala
package tools.packet.serverlist

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GetServerStatus(private var _status: ServerStatus) extends PacketInput {
  def status: ServerStatus = _status

  def this(serverStatusId: Int) = {
    this(ServerStatus.fromValue(serverStatusId))
  }

  override def opcode(): SendOpcode = SendOpcode.SERVER_STATUS
}

File: engine/engine-model/src/main/java/tools/packet/serverlist/GetServerStatus.java
package tools.packet.serverlist;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GetServerStatus(ServerStatus status) implements PacketInput {
   public GetServerStatus(Integer status) {
      this(ServerStatus.fromValue(status));
   }

   @Override
   public SendOpcode opcode() {
      return SendOpcode.SERVER_STATUS;
   }
}